battery/imx6ull-vbat/imx6ull-vbat.c

```diff
@@ -0,0 +1,446 @@
+/*
+ * Phoenix-RTOS
+ *
+ * IMX6ULL battery driver.
+ *
+ * Reads battery voltage
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Ziemowit Leszczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+#include <sys/minmax.h>
+
+#include <sys/msg.h>
+#include <sys/mman.h>
+#include <sys/platform.h>
+#include <posix/utils.h>
+
+#include <phoenix/arch/imx6ull.h>
+
+#define LOG_TAG       "imx6ull-vbat: "
+#define log(fmt, ...) printf(LOG_TAG fmt "\n", ##__VA_ARGS__)
+
+#define DEV_VBAT_ID   0
+#define DEV_VBAT_NAME "vbat"
+
+#define GPIO1_BASE_ADDR 0x209C000
+#define GPIO5_BASE_ADDR 0x20AC000
+
+#define VBAT_GPIO_MUX pctl_mux_gpio1_04
+#define VBAT_GPIO_PAD pctl_pad_gpio1_04
+#define VBAT_GPIO_PIN 4
+
+#define VBAT_ON_GPIO_PIN 9
+
+#define ADC_BASE_ADDR 0x2198000
+#define ADC_CLK       pctl_clk_adc1
+#define ADC_CHANNEL   4
+#define ADC_VREF      3.5f
+#define ADC_MAX_VALUE 4095
+
+typedef volatile struct {
+	uint32_t DR;
+	uint32_t GDIR;
+	uint32_t PSR;
+	uint32_t ICR1;
+	uint32_t ICR2;
+	uint32_t IMR;
+	uint32_t ISR;
+	uint32_t EDGE_SEL;
+} gpio_t;
+
+typedef volatile struct {
+	uint32_t HC0;
+	uint32_t reserved0;
+	uint32_t HS;
+	uint32_t R0;
+	uint32_t reserved1;
+	uint32_t CFG;
+	uint32_t GC;
+	uint32_t GS;
+	uint32_t CV;
+	uint32_t OFS;
+	uint32_t CAL;
+} adc_t;
```

Comment:
Please stick to the current code style, i.e. `*(base + offset)`. I'm not saying that your approach is inferior in any way, I'd only like to preserve consistency within the codebase